misc/python/materialize/mzbuild.py

```diff
@@ -836,10 +837,30 @@ def ensure(self) -> None:
         """
         deps_to_build = [dep for dep in self if not dep.is_published_if_necessary()]
         self._prepare_batch(deps_to_build)
+
+        images_to_push = []
         for dep in deps_to_build:
             dep.build()
             if dep.publish:
-                spawn.runv(["docker", "push", dep.spec()])
+                images_to_push.append(dep.spec())
+
+        # Push all Docker images in parallel to minimize build time.
```

Comment:
That was easier than expected. I was looking for a way to have a single `docker push` call upload multiple images, went in the wrong direction :)